### PR TITLE
Fix cache line size detection on power

### PR DESCRIPTION
**Pull request by nysal**
Due to the nature of the cache architecture on power,
we don't export coherency_line_size for L2 in sysfs.
If we are unable to get the L2 cache line size, try L1.

See open-mpi/ompi#383 for more information.

### CODE REVIEW

**Comment by mellanox-github** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://bgate.mellanox.com/jenkins/job/gh-ompi-master-pr/220/
Test PASSed.

File: ompi/mca/coll/libnbc/coll_libnbc_component.c [hunk 1]
```diff
@@ -97,7 +98,6 @@ libnbc_open(void)
                               NULL);
     if (OMPI_SUCCESS != ret) return ret;
 
-    OBJ_CONSTRUCT(&mca_coll_libnbc_component.active_requests, opal_list_t);
```
Comment (jsquyres) [inline ompi/mca/coll/libnbc/coll_libnbc_component.c]: This fix is unrelated to the L2/L1 cache size fix, right?

Comment (nysal) [inline ompi/mca/coll/libnbc/coll_libnbc_component.c]: Correct, if the free list initialization fails (due to the cache line size bug), then ompi crashes in libnbc_close() as the mca_coll_libnbc_component.active_requests is not initialized. This change justs moves the initialization to an earlier location so that it doesn't crash in libnbc_close().  This kind of misdirected me early on from the original issue.

Comment (jsquyres) [inline ompi/mca/coll/libnbc/coll_libnbc_component.c]: Would it be terrible if you split this out into a separate commit?  Can be on this same PR, but just logically, it has nothing to do with this commit (and the commit message you provided).

Thanks!

**Comment by mellanox-github** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://bgate.mellanox.com/jenkins/job/gh-ompi-master-pr/221/

**Build Log**
*last 50 lines*

 ```
[...truncated 4922 lines...]
make[2]: Entering directory `/scrap/jenkins/jenkins/jobs/gh-ompi-master-pr/workspace/opal/include'
make[3]: Entering directory `/scrap/jenkins/jenkins/jobs/gh-ompi-master-pr/workspace/opal/include'
make[3]: Nothing to be done for `install-exec-am'.
 /bin/mkdir -p '/var/lib/jenkins/jobs/gh-ompi-master-pr/workspace/ompi_install1/include/openmpi'
 /bin/mkdir -p '/var/lib/jenkins/jobs/gh-ompi-master-pr/workspace/ompi_install1/include/openmpi'
 /usr/bin/install -c -m 644  opal_config.h '/var/lib/jenkins/jobs/gh-ompi-master-pr/workspace/ompi_install1/include/openmpi/.'
 /bin/mkdir -p '/var/lib/jenkins/jobs/gh-ompi-master-pr/workspace/ompi_install1/include/openmpi/opal/sys/amd64'
 /usr/bin/install -c -m 644  opal/sys/amd64/atomic.h opal/sys/amd64/timer.h '/var/lib/jenkins/jobs/gh-ompi-master-pr/workspace/ompi_install1/include/openmpi/opal/sys/amd64'
 /bin/mkdir -p '/var/lib/jenkins/jobs/gh-ompi-master-pr/workspace/ompi_install1/include/openmpi/opal'
 /usr/bin/install -c -m 644  opal/version.h '/var/lib/jenkins/jobs/gh-ompi-master-pr/workspace/ompi_install1/include/openmpi/opal'
 /bin/mkdir -p '/var/lib/jenkins/jobs/gh-ompi-master-pr/workspace/ompi_install1/include/openmpi/opal/sys'
 /usr/bin/install -c -m 644  opal/sys/architecture.h opal/sys/atomic.h opal/sys/atomic_impl.h opal/sys/timer.h opal/sys/cma.h '/var/lib/jenkins/jobs/gh-ompi-master-pr/workspace/ompi_install1/include/openmpi/opal/sys'
 /bin/mkdir -p '/var/lib/jenkins/jobs/gh-ompi-master-pr/workspace/ompi_install1/include/openmpi/opal/sys/sparcv9'
 /usr/bin/install -c -m 644  opal/sys/sparcv9/atomic.h opal/sys/sparcv9/timer.h '/var/lib/jenkins/jobs/gh-ompi-master-pr/workspace/ompi_install1/include/openmpi/opal/sys/sparcv9'
 /bin/mkdir -p '/var/lib/jenkins/jobs/gh-ompi-master-pr/workspace/ompi_install1/include/openmpi/opal/sys/alpha'
 /usr/bin/install -c -m 644  opal/sys/alpha/atomic.h '/var/lib/jenkins/jobs/gh-ompi-master-pr/workspace/ompi_install1/include/openmpi/opal/sys/alpha'
 /bin/mkdir -p '/var/lib/jenkins/jobs/gh-ompi-master-pr/workspace/ompi_install1/include/openmpi/opal/sys/ia64'
 /usr/bin/install -c -m 644  opal/sys/ia64/atomic.h opal/sys/ia64/timer.h '/var/lib/jenkins/jobs/gh-ompi-master-pr/workspace/ompi_install1/include/openmpi/opal/sys/ia64'
 /bin/mkdir -p '/var/lib/jenkins/jobs/gh-ompi-master-pr/workspace/ompi_install1/include/openmpi/opal/sys/osx'
 /usr/bin/install -c -m 644  opal/sys/osx/atomic.h '/var/lib/jenkins/jobs/gh-ompi-master-pr/workspace/ompi_install1/include/openmpi/opal/sys/osx'
 /bin/mkdir -p '/var/lib/jenkins/jobs/gh-ompi-master-pr/workspace/ompi_install1/include/openmpi/opal/sys/arm'
 /usr/bin/install -c -m 644  opal/sys/arm/atomic.h opal/sys/arm/timer.h '/var/lib/jenkins/jobs/gh-ompi-master-pr/workspace/ompi_install1/include/openmpi/opal/sys/arm'
 /bin/mkdir -p '/var/lib/jenkins/jobs/gh-ompi-master-pr/workspace/ompi_install1/include/openmpi/opal/sys/mips'
 /usr/bin/install -c -m 644  opal/sys/mips/atomic.h opal/sys/mips/timer.h '/var/lib/jenkins/jobs/gh-ompi-master-pr/workspace/ompi_install1/include/openmpi/opal/sys/mips'
 /usr/bin/install -c -m 644  opal_config_top.h opal_config_bottom.h opal_stdint.h '/var/lib/jenkins/jobs/gh-ompi-master-pr/workspace/ompi_install1/include/openmpi/.'
 /bin/mkdir -p '/var/lib/jenkins/jobs/gh-ompi-master-pr/workspace/ompi_install1/include/openmpi/opal/sys/sync_builtin'
 /usr/bin/install -c -m 644  opal/sys/sync_builtin/atomic.h '/var/lib/jenkins/jobs/gh-ompi-master-pr/workspace/ompi_install1/include/openmpi/opal/sys/sync_builtin'
 /bin/mkdir -p '/var/lib/jenkins/jobs/gh-ompi-master-pr/workspace/ompi_install1/include/openmpi/opal/sys/powerpc'
 /usr/bin/install -c -m 644  opal/sys/powerpc/atomic.h opal/sys/powerpc/timer.h '/var/lib/jenkins/jobs/gh-ompi-master-pr/workspace/ompi_install1/include/openmpi/opal/sys/powerpc'
 /bin/mkdir -p '/var/lib/jenkins/jobs/gh-ompi-master-pr/workspace/ompi_install1/include/openmpi/opal/sys/ia32'
 /usr/bin/install -c -m 644  opal/sys/ia32/atomic.h opal/sys/ia32/timer.h '/var/lib/jenkins/jobs/gh-ompi-master-pr/workspace/ompi_install1/include/openmpi/opal/sys/ia32'
 /bin/mkdir -p '/var/lib/jenkins/jobs/gh-ompi-master-pr/workspace/ompi_install1/include/openmpi/opal'
 /usr/bin/install -c -m 644  opal/align.h opal/constants.h opal/opal_socket_errno.h opal/types.h opal/prefetch.h opal/hash_string.h opal/frameworks.h opal/opal_portable_platform.h '/var/lib/jenkins/jobs/gh-ompi-master-pr/workspace/ompi_install1/include/openmpi/opal'
make[3]: Leaving directory `/scrap/jenkins/jenkins/jobs/gh-ompi-master-pr/workspace/opal/include'
make[2]: Leaving directory `/scrap/jenkins/jenkins/jobs/gh-ompi-master-pr/workspace/opal/include'
Making install in libltdl
make[2]: Entering directory `/scrap/jenkins/jenkins/jobs/gh-ompi-master-pr/workspace/opal/libltdl'
CDPATH="${ZSH_VERSION+.}:" && cd . && /bin/sh /scrap/jenkins/jenkins/jobs/gh-ompi-master-pr/workspace/config/missing aclocal-1.15 -I ../../config
aclocal-1.15: error: ../../config/autogen_found_items.m4:316: file 'opal/mca/backtrace/configure.m4' does not exist
make[2]: *** [aclocal.m4] Error 1
make[2]: Leaving directory `/scrap/jenkins/jenkins/jobs/gh-ompi-master-pr/workspace/opal/libltdl'
make[1]: *** [install-recursive] Error 1
make[1]: Leaving directory `/scrap/jenkins/jenkins/jobs/gh-ompi-master-pr/workspace/opal'
make: *** [install-recursive] Error 1
+ exit 10
Build step 'Execute shell' marked build as failure
[BFA] Scanning build for known causes...

[BFA] Done. 0s
```

Test FAILed.

**Comment by nysal** [comment]
Jeff, any idea what the problem is? The branch builds fine for me. The single commit which was submitted earlier passed. The code is the same, its just split into two commits now.

**Comment by jsquyres** [comment]
Mellanox upgraded Libtool yeterday, which caused some problems.  Let's just see if it builds now.

retest this please

**Comment by jsquyres** [comment]
retest this please

**Comment by mellanox-github** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://bgate.mellanox.com/jenkins/job/gh-ompi-master-pr/236/
Test PASSed.

**Comment by nysal** [comment]
@jsquyres : just checking if there are any other comments on this PR

**Comment by jsquyres** [comment]
:+1:

**Comment by nysal** [comment]
Can you please push the patch to master branch? I dont think I have commit rights

**Comment by jsquyres** [comment]
Ah!  Sorry, I thought you did, and was wondering why you didn't finish the merge.  :-)

Your name is still on the last schedule A that I have from IBM, so I just gave you commit rights to the OMPI repo.

I note that this PR can no longer be automatically merged -- this typically means that you need to rebase to the head of master to fix a conflict that has come up since you created it.

**Comment by nysal** [comment]
Thanks Jeff. I'll rebase with the master branch.

**Comment by mellanox-github** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://bgate.mellanox.com/jenkins/job/gh-ompi-master-pr/286/
Test PASSed.
